libq/hash.c

```diff
@@ -354,3 +354,34 @@ hash_file_at_cb(int pfd, const char *fname, int hash, hash_cb_t cb)
 
 	return _hash_file_buf;
 }
+
+
+/*
+*Generate and alloc in heap a hash with MD5.
```

Comment:
instead of alloc, you can perhaps reuse the _hash_file_buf that's static for this purpose?